prov/efa/src/rxr/rxr_pkt_type_misc.c

```diff
@@ -643,10 +643,22 @@ void rxr_pkt_handle_receipt_recv(struct rxr_ep *ep,
 	if (!tx_entry) {
 		FI_WARN(&rxr_prov, FI_LOG_CQ,
 			"Failed to retrive the tx_entry when hadling receipt packet.\n");
-		return;
 	}
 
-	rxr_cq_handle_tx_completion(ep, tx_entry);
+	tx_entry->rxr_flags |= RXR_RECEIPT_HANDLED;
+	if (tx_entry->bytes_acked == 0 ||
+	    tx_entry->bytes_acked == tx_entry->total_len) {
+		/*
+		 * For long message protocol, when FI_DELIVERY_COMPLETE
```

Comment:
Maybe  add a note here that for eager and medium DC message, bytes_acked is not updated so is always 0.

Reply:
Note added, resolving.